test/module/irohad/torii/query_service_test.cpp

```diff
@@ -29,54 +29,112 @@ using namespace shared_model::detail;
 using ::testing::_;
 using ::testing::Return;
+using ::testing::Invoke;
+using ::testing::Truly;
 
 class QueryServiceTest : public ::testing::Test {
  public:
   void SetUp() override {
     query_processor = std::make_shared<MockQueryProcessor>();
     // any query
-    query = shared_model::proto::QueryBuilder()
-                .creatorAccountId("user@domain")
-                .createdTime(iroha::time::now())
-                .queryCounter(1)
-                .getAccountTransactions("user@domain")
-                .build()
-                .signAndAddSignature(
-                    shared_model::crypto::DefaultCryptoAlgorithmType::
-                        generateKeypair())
-                .getTransport();
+    query = std::make_shared<shared_model::proto::Query>(
+        shared_model::proto::QueryBuilder()
+            .creatorAccountId("user@domain")
+            .createdTime(iroha::time::now())
+            .queryCounter(1)
+            .getAccount("user@domain")
+            .build()
+            .signAndAddSignature(
+                shared_model::crypto::DefaultCryptoAlgorithmType::
+                    generateKeypair()));
+
+    // TODO: IR-1041 Update to query response builders (kamilsa, 04.03.2018)
+    protocol::QueryResponse response;
+    response.set_query_hash(
+        shared_model::crypto::toBinaryString(query->hash()));
+    auto account_response = response.mutable_account_response();
+    account_response->add_account_roles("user");
+    auto account = account_response->mutable_account();
+    account->set_domain_id("ru");
+    account->set_account_id("a");
+    account->set_quorum(2);
+
+    model_response = std::make_shared<shared_model::proto::QueryResponse>(
+        std::move(response));
   }
 
   void init() {
     query_service = std::make_shared<QueryService>(query_processor);
   }
 
-  protocol::Query query;
-  protocol::QueryResponse response;
+  std::shared_ptr<shared_model::proto::Query> query;
+  std::shared_ptr<shared_model::proto::QueryResponse> model_response;
   std::shared_ptr<QueryService> query_service;
   std::shared_ptr<MockQueryProcessor> query_processor;
 };
 
-TEST_F(QueryServiceTest, SubscribeQueryProcessorWhenInit) {
-  // query service is subscribed to query processor
+/**
+ * @given query and expected valid response
+ * @when query is sent to query service and query_processor processes query
+ * @then expected response is returned
+ */
+TEST_F(QueryServiceTest, ValidWhenUniqueHash) {
+  // unique query => query handled by query processor
+  rxcpp::subjects::subject<
+      std::shared_ptr<shared_model::interface::QueryResponse>>
+      notifier;
   EXPECT_CALL(*query_processor, queryNotifier())
-      .WillOnce(
-          Return(rxcpp::observable<>::empty<std::shared_ptr<QueryResponse>>()));
-
+      .WillOnce(Return(notifier.get_observable()));
+  EXPECT_CALL(*query_processor,
+              queryHandle(Truly(
+                  [this](std::shared_ptr<shared_model::interface::Query> rhs) {
```

Comment:
Note, without Truly and this expression, GMOCK would match query by comparing two `shared_ptr`s using `operator==` which for `shared_ptr` compares `lhs.get() == rhs.get()` i.e. pointer values

Reply:
mb leave comment in the code?